### PR TITLE
FIx/Allow object removal without link object

### LINKED ISSUE #2100: Complex object cannot be removed if link object is deleted
Let say we have:
1. Complex object with id `123`
2. Link object with id `123-link`
3. Link object with id `123-link` was deleted

Steps:
1. Try to delete complex object

```
neofs-cli --config wallet_config.yml object delete --rpc-endpoint '<endpoint>' --wallet '51bb4c44-f545-4967-83e7-d36eb166e513.json' --cid '<container_id>' --oid '123'
```

## Expected Behavior
It should be possible to still delete object

## Current Behavior
`
failed to get linking object's header: read object header via client: status: code = 2052 message = object already removed`

## Tests
in feature branch `feature-abereziny-test-complex-deletion-with-no-link`
`pytest_tests/testsuites/object/test_object_lock.py#test_should_be_possible_to_delete_complex_object_if_link_object_is_absent`

### PR DESCRIPTION
**Pull request by carpawell**
Closes #2100.

### CODE REVIEW

**Comment by sami-nspcc** [comment]
I am running integration tests

**Comment by sami-nspcc** [comment]
I have recieved a notification of a new pull request. I am starting the build of images and binaries for further testing. Build number is 3774

**Comment by sami-nspcc** [comment]
Test run is finished. Please download the tarball from <a href="https://http.t5.fs.neo.org/get/FszwwAx3VznDysQGxBUMRj1zj3zp8VcohorVDnunrviV/FZMsHDfjZTf8VT5gqWAhZQTYmiHPzkGEXYxg5a1tueXu">link</a>. Untar and use `allure open` to watch the report

**Comment by sami-nspcc** [comment]
I am running integration tests

**Review by cthulhu-rider — APPROVED** [review]

File: cmd/neofs-cli/modules/object/util.go [hunk 1]
```diff
@@ -381,14 +381,18 @@ func collectObjectRelatives(cmd *cobra.Command, cli *client.Client, cnr cid.ID,
...
-		common.PrintVerbose("Received split members from the linking object: %v", children)
+			// include linking object
+			return append(children, idLinking)
+		}
 
-		// include linking object
-		return append(children, idLinking)
+		// linking object is not required for
+		// object collecting
+		common.PrintVerbose("failed to get linking object's header: %w", err)
```
Comment (fyrchik) [inline cmd/neofs-cli/modules/object/util.go:395]: What is the error was `context deadline exceeded`? Is it possible that linking object will never be removed?

Comment (carpawell) [reply cmd/neofs-cli/modules/object/util.go:395]: do you mean attaching linking object anyway despite its accessibility?

**Review by fyrchik — APPROVED** [review]

**Pull request merged**